neurots/generate/algorithms/tmdgrower.py

```diff
@@ -67,15 +67,19 @@ def __init__(
         self.persistence_length = self.barcode.get_persistence_length()
         # Validate parameters and distributions
         if not skip_validation:
-            # Consistency check between parameters - persistence diagram
-            barSZ = input_data["min_bar_length"]
-            stepSZ = params["step_size"]["norm"]["mean"]
-            if stepSZ >= barSZ:
-                L.warning(
-                    "Selected step size %f is too big for bars of size %f",
-                    stepSZ,
-                    barSZ,
-                )
+            self.check_min_bar_length(params, input_data)
+
+    @staticmethod
+    def check_min_bar_length(params, distrs):
```

Comment:
Hmm, if the checks need more info than the parames and distrs, you could have the entire grower passing to each check function to ensure that for all check cases you will always have all the info available.

Having the check attached to the grower however makes it unavailable to use it from another grower for example that may not derive from this one. And it would be strange to import this grower just to use its check statics.

Reply:
Yeah ok, it's probably better to get rid of the inheritance limitations indeed. I moved this.